lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -869,6 +869,81 @@ namespace Js
     }
 
     template <typename T>
+    BOOL DictionaryTypeHandlerBase<T>::DeleteProperty(DynamicObject *instance, JavascriptString *propertyNameString, PropertyOperationFlags propertyOperationFlags)
+    {
+        AssertMsg(!PropertyRecord::IsPropertyNameNumeric(propertyNameString->GetString(), propertyNameString->GetLength()),
+            "Numeric property names should have been converted to uint or PropertyRecord* ");
+
+        ScriptContext* scriptContext = instance->GetScriptContext();
+        DictionaryPropertyDescriptor<T>* descriptor;
+        JsUtil::CharacterBuffer<WCHAR> propertyName(propertyNameString->GetString(), propertyNameString->GetLength());
+
+        if (propertyMap->TryGetReference(propertyName, &descriptor))
+        {
+            Assert(descriptor->SanityCheckFixedBits());
+
+            if (descriptor->Attributes & PropertyDeleted)
+            {
+                return true;
```

Comment:
Yes, it is intentional. allowLetConstGlobal is true only when dealing with a property on the root object. However, if we are coming to this function, the object is not a root object. Does that make sense? I was a little skeptical of this part, so I appreciate a second set of eyes.

---
In reply to: [78791249](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78791249) [](ancestors = 78791249)